sqlglot/dialects/starrocks.py

```diff
@@ -13,11 +13,58 @@
 )
 from sqlglot.dialects.mysql import MySQL
 from sqlglot.helper import seq_get
+from sqlglot.tokens import TokenType
+
+
+def _duplicate_key_sql(self, expression: exp.DuplicateKeyProperty) -> str:
+    expressions = self.expressions(expression, flat=True)
+    options = self.expressions(expression, key="options", flat=True, sep=" ")
+    options = f" {options}" if options else ""
+    return f"DUPLICATE KEY ({expressions}){options}"
+
+
+def _distributed_by_hash_sql(self, expression: exp.DistributedByHashProperty) -> str:
+    expressions = self.expressions(expression, key="expressions", flat=True)
+    sorted_by = self.expressions(expression, key="sorted_by", flat=True)
+    sorted_by = f" ORDER BY ({sorted_by})" if sorted_by else ""
+    buckets = self.sql(expression, "buckets")
+    # Since StarRocks v2.5.7, the number of buckets is AUTO by default.
+    # https://docs.starrocks.io/docs/sql-reference/sql-statements/data-definition/CREATE_TABLE/#distribution_desc
+    if expression.auto_bucket:
+        buckets = "AUTO"
+    # [DISTRIBUTED BY HASH (k1[,k2 ...]) [BUCKETS num] [ORDER BY (k1[,k2 ...])]]
+    return f"DISTRIBUTED BY HASH ({expressions}) BUCKETS {buckets}{sorted_by}"
+
+
+def _distributed_by_random_sql(self, expression: exp.DistributedByRandomProperty) -> str:
+    expressions = self.expressions(expression, flat=True)
+    sorted_by = self.expressions(expression, key="sorted_by", flat=True)
+    sorted_by = f" ORDER BY ({sorted_by})" if sorted_by else ""
+    buckets = self.sql(expression, "buckets")
+    if expression.auto_bucket:
+        buckets = "AUTO"
+    # [DISTRIBUTED BY RANDOM (k1[,k2 ...]) [BUCKETS num] [ORDER BY (k1[,k2 ...])]]
+    return f"DISTRIBUTED BY RANDOM ({expressions}) BUCKETS {buckets}{sorted_by}"
+
+
+def _property_sql(self: MySQL.Generator, expression: exp.Property) -> str:
+    # [PROPERTIES (['key1'='value1', 'key2'='value2', ...])]
+    return f"{self.property_name(expression, string_key=True)}={self.sql(expression, 'value')}"
 
 
 class StarRocks(MySQL):
     STRICT_JSON_PATH_SYNTAX = False
 
+    class Tokenizer(MySQL.Tokenizer):
+        KEYWORDS = {
+            **MySQL.Tokenizer.KEYWORDS,
+            "DECIMAL64": TokenType.DECIMAL,
+            "DECIMAL128": TokenType.BIGDECIMAL,
```

Comment:
We need to test these mappings. Can you share docs to justify these entries?

Reply:
Oh ok, I'll add it here.
```
- Several types of decimal: https://docs.starrocks.io/docs/unloading/Flink_connector/#data-type-mapping-between-starrocks-and-flink
> Focus on the starrocks column

- Use:
https://docs.starrocks.io/docs/sql-reference/sql-functions/math-functions/round/#parameters
```

I will also add ut to test these mappings

Reply:
According to the documentation, I additionally added `DECIMAL32`

Reply:
In starrocks, it is executed as follows:
![image](https://github.com/user-attachments/assets/f3098d73-8381-4588-b0b0-12cd77a1b102)